components/script/dom/mouseevent.rs

```diff
@@ -157,7 +157,51 @@ impl MouseEvent {
         can_gc: CanGc,
     ) -> DomRoot<MouseEvent> {
         let ev = MouseEvent::new_uninitialized_with_proto(window, proto, can_gc);
-        ev.InitMouseEvent(
+        ev.initialize_mouse_event(
+            type_,
+            can_bubble,
+            cancelable,
+            view,
+            detail,
+            screen_x,
+            screen_y,
+            client_x,
+            client_y,
+            ctrl_key,
+            alt_key,
+            shift_key,
+            meta_key,
+            button,
+            buttons,
+            related_target,
+            point_in_target,
+        );
+        ev
+    }
+
+    // https://w3c.github.io/uievents/#initialize-a-mouseevent
+    #[allow(clippy::too_many_arguments)]
+    pub fn initialize_mouse_event(
+        &self,
+        type_: DOMString,
+        can_bubble: EventBubbles,
+        cancelable: EventCancelable,
+        view: Option<&Window>,
+        detail: i32,
+        screen_x: i32,
+        screen_y: i32,
+        client_x: i32,
+        client_y: i32,
+        ctrl_key: bool,
+        alt_key: bool,
+        shift_key: bool,
+        meta_key: bool,
+        button: i16,
+        buttons: u16,
+        related_target: Option<&EventTarget>,
+        point_in_target: Option<Point2D<f32>>,
+    ) {
+        self.InitMouseEvent(
```

Comment:
This one is now deprecated, so we may want to file an issue to remove it, instead of starting to rely on it for other UI events. 

https://w3c.github.io/uievents/#dom-mouseevent-initmouseevent


I think the same goes for `InitUIEvent`. 

Would be nice to restructure the code to actually follow the spec using the private methods, and eventually calling into the DOM spec `init_event`, with links everywhere. 

Good for filing an issue for a follow-up, so that we don't have features on top of deprecated infrastructure.

Reply:
I added a TODO comment and will open a follow up issue. But I'm not sure the scope of the meta issue regarding clean-up UI event infra. Might need your feedback there.